libass/ass_library.c

```diff
@@ -95,24 +95,13 @@ void ass_set_style_overrides(ASS_Library *priv, char **list)
         *q = strdup(*p);
 }
 
-static int grow_array(void **array, int nelem, size_t elsize)
-{
-    if (!(nelem & 31)) {
-        void *ptr = realloc(*array, (nelem + 32) * elsize);
-        if (!ptr)
-            return 0;
-        *array = ptr;
-    }
-    return 1;
-}
-
 void ass_add_font(ASS_Library *priv, const char *name, const char *data, int size)
 {
-    int idx = priv->num_fontdata;
+    size_t idx = priv->num_fontdata;
     if (!name || !data || !size)
         return;
-    if (!grow_array((void **) &priv->fontdata, priv->num_fontdata,
-                    sizeof(*priv->fontdata)))
+    if (!(idx & 31) && !(idx & (idx - 1)) && // power of two >= 32, or zero --> realloc time
```

Comment:
Can be replaced with `!(idx & (idx - 32))`.

Reply:
Done, thx.